jack.py

```diff
@@ -739,12 +739,21 @@ def set_process_callback(self, callback, userdata=None):
             is called.
 
         """
-        @self._callback("JackProcessCallback", error=STOP_CALLING)
-        def callback_wrapper(frames, _):
-            return callback(frames, userdata)
+        if isinstance(callback, _ffi.CData):
+            if _ffi.typeof(callback) != _ffi.typeof("JackProcessCallback"):
+                raise ValueError("Expected %s, got %s" % \
+                                 (_ffi.typeof("JackProcessCallback"),
+                                  _ffi.typeof(callback)))
```

Comment:
I know, but I think this isn't worth the effort for such an extremely uncommon use case.

The interpreter will throw an exception anyway (and it will not crash!) when it tries to call the thing.

I would only create a custom exception if its a very common error (which this is not) or when continuing would lead to nasty or subtle or hard-to-find errors later in the program (which this does not, because an exception is thrown by the interpreter and the process callback will not be called again).